cli/src/test/kotlin/AuthLoginCommandTest.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2025 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import com.charleskorn.kaml.Yaml
+import com.charleskorn.kaml.decodeFromStream
+
+import com.github.ajalt.clikt.command.test
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldBe
+
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.mockkConstructor
+import io.mockk.mockkStatic
+import io.mockk.unmockkAll
+
+import kotlin.io.path.createTempDirectory
+
+import org.eclipse.apoapsis.ortserver.cli.OrtServerMain
+import org.eclipse.apoapsis.ortserver.cli.utils.configDir
+import org.eclipse.apoapsis.ortserver.client.auth.AuthService
+import org.eclipse.apoapsis.ortserver.client.auth.AuthenticationStorage
+import org.eclipse.apoapsis.ortserver.client.auth.TokenInfo
+import org.eclipse.apoapsis.ortserver.client.auth.Tokens
+
+class AuthLoginCommandTest : StringSpec({
+    afterEach { unmockkAll() }
+
+    "Auth login command" should {
+        "store the authentication information in a local file" {
+            val command = OrtServerMain()
+
+            val testDir = createTempDirectory("osc").toFile()
+            val testAuthFile = testDir.resolve("auth.yml")
+            mockkStatic("org.eclipse.apoapsis.ortserver.cli.utils.ConfigDirHelperKt")
+            every { configDir } returns testDir
+
+            mockkConstructor(AuthService::class)
+            coEvery { anyConstructed<AuthService>().generateToken("testUser", "testPassword") } returns TokenInfo(
+                accessToken = "testAccessToken",
+                refreshToken = "testRefreshToken",
+                expiresInSeconds = 3600
+            )
+
+            val result = command.test(
+                listOf(
+                    "auth",
+                    "login",
+                    "--base-url", "http://localhost:8080/",
+                    "--token-url", "http://localhost/token",
+                    "--username", "testUser",
+                    "--password", "testPassword",
+                    "--client-id", "test-client-id"
+                )
+            )
+
+            Yaml.default.decodeFromStream<AuthenticationStorage>(testAuthFile.inputStream()).let {
```

Comment:
Also here the `InputStream` should be closed.

Reply:
Prefer something like `with()` or `also {}` over `let {}` if the return value is not used.

Reply:
Obsolete with the `AuthenticationStorage` refactoring.